pkg/types/ccipocr3/chain_accessor.go

```diff
@@ -0,0 +1,313 @@
+	// based on their timestamp.
+	//
+	// Access Type: Event(CCIPMessageSent)
+	// Contract: OnRamp
+	// Confidence: Finalized
+	MsgsBetweenSeqNums(
+		ctx context.Context,
+		dest ChainSelector,
+		seqNumRange SeqNumRange,
+	) ([]Message, error)
+
+	// LatestMsgSeqNum returns the sequence number associated with the most
+	// recent message being sent to a given destination.
+	//
+	// Access Type: Event(CCIPMessageSent)
+	// Contract: OnRamp
+	// Confidence: Finalized
+	//
+	// TODO: Rename function in CAL to LatestMessageTo(dest) and return the entire message.
+	//       See Design Doc (LatestMsgSeqNum) for notes.
+	LatestMsgSeqNum(ctx context.Context, dest ChainSelector) (SeqNum, error)
+
+	// GetExpectedNextSequenceNumber returns the expected next sequence number
+	// messages being sent to the provided destination.
+	//
+	// Access Type: Method(GetExpectedNextSequenceNumber)
+	// Contract: OnRamp
+	// Confidence: Unconfirmed
+	GetExpectedNextSequenceNumber(
+		ctx context.Context,
+		dest ChainSelector,
+	) (SeqNum, error)
+
+	// GetTokenPriceUSD looks up a token price in USD. The address value should
+	// be retrieved from a configuration cache (i.e. ConfigPoller).
+	//
+	// Access Type: Method(GetTokenPRice)
+	// Contract: FeeQuoter
+	// Confidence: Unconfirmed
+	//
+	// Notes: This function is new and serves as a general price interface for
+	//        both LinkPriceUSD and GetWrappedNativeTokenPriceUSD.
+	//        See Design Doc (Combined Token Price Helper) for notes.
+	GetTokenPriceUSD(
+		ctx context.Context,
+		address UnknownAddress,
+	) (BigInt, error)
+
+	// GetFeeQuoterDestChainConfig returns the fee quoter destination chain config.
+	//
+	// Access Type: Method(GetDestChainConfig)
+	// Contract: FeeQuoter
+	// Confidence: Unconfirmed
+	//
+	// Notes: This is a new general purpose function needed to implement
+	//        GetMedianDataAvailabilityGasConfig.
+	GetFeeQuoterDestChainConfig(
+		ctx context.Context,
+		dest ChainSelector,
+	) (FeeQuoterDestChainConfig, error)
+}
+
+type RMNAccessor interface {
+	// GetRMNRemoteConfig
+	//
+	// TODO: This function should be removed and replaced with direct access to
+	//       the ConfigPoller. It also manages chain reader contract binding
+	//       directly instead of using DiscoverContracts.
+	GetRMNRemoteConfig(ctx context.Context) (rmntypes.RemoteConfig, error)
+
+	// GetRmnCurseInfo returns rmn curse/pausing information about the provided chains
+	// from the destination chain RMN remote contract. Caller should be able to access destination.
+	GetRmnCurseInfo(ctx context.Context) (CurseInfo, error)
+}
+
+////////////////////////////////////////////////////////////////
+// TODO: Find a better location for the types below this line //
+//       For the purpose of designing these interfaces, the   //
+//       location is not critical.                            //
+////////////////////////////////////////////////////////////////
```

Comment:
Open to suggestions on types. I don't hate defining all of these in the ccipocr3 package.